code/modules/mob/living/simple_animal/friendly/mouse.dm

```diff
@@ -87,26 +87,64 @@
 		if(prob(5))
 			to_chat(src, "<span class = 'warning'>You are getting hungry!</span>")
 
-	find_nearby_disease()//getting diseases from
+	handle_body_temperature()//I bestow upon mice the gift of thermoregulation, so they can handle the fever caused by disease.
 
-	if(SSair.current_cycle%4==2)//Only try to breath diseases every 4 seconds
-		breath_airborne_diseases()
+	//------------------------DISEASE STUFF--------------------------------------------------------
+	if(!locked_to || !istype(locked_to,/obj/item/critter_cage))//cages isolate from diseases
+		find_nearby_disease()//getting diseases from
 
-	for (var/mob/living/simple_animal/mouse/M in range(1,src))
-		share_contact_diseases(M)
+		if(SSair.current_cycle%4==2)//Only try to breath diseases every 4 seconds
+			breath_airborne_diseases()
+
+		for (var/mob/living/simple_animal/mouse/M in range(1,src))
+			share_contact_diseases(M)
+
+		if (virus2.len)
```

Comment:
I've seen this bit of code twice in a living mob now. I'm not certain since I'm not familiar with living code but would it not be possible to put this in mob/living then add an affected_by_diseases variable that defaults to 0?